### PR TITLE
update plaid-go to match 2019-05-29 API release

### PR DESCRIPTION
**Pull request by tanatplaid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,8 @@ Each major version of `plaid-go` targets a specific version of the Plaid API:
 
 | API version | plaid-go release |
 | ----------- | ------------------ |
-| [`2018-05-22`][api-version-2018-05-22] (**latest**) | `1.x.x` |
+| [`2019-05-29`][api-version-2019-05-29] (**latest**) | `1.x.x` |
```
Comment (jasonjia93) [inline README.md]: I think we should upgrade the version to `2.x.x`

File: plaid/accounts.go [hunk 2]
```diff
@@ -30,6 +30,18 @@ type AccountNumber struct {
...
 }
 
+type IBANNumber struct {
+	AccountID string `json:"account_id"`
+	IBAN      string `json:"iban"`
+	BIC       string `json:"bic"`
+}
+
+type BACSNumber struct {
+	BACS     string `json:"bacs"`
```
Comment (jasonjia93) [inline plaid/accounts.go]: This should be `AccountID`, not `BACS`

File: plaid/identity.go [hunk 3]
```diff
@@ -19,10 +19,11 @@ type Address struct {
 }
 
 type AddressData struct {
-	City   string `json:"city"`
-	State  string `json:"state"`
-	Street string `json:"street"`
-	Zip    string `json:"zip"`
+	City       string `json:"city"`
```
Comment (ntindall) [inline plaid/identity.go]: we need to add `AssetsAddressData` - the change does not apply to `AssetReport` json.

Comment (tanatplaid) [reply plaid/identity.go]: the current assets identity and transactions data is not under the assets.go. should we add it in?

Comment (ntindall) [reply plaid/identity.go]: Good catch! Lets keep it as a separate change from this PR just to keep the scope isolated.

**Pull request merged**